### PR TITLE
Improve handling of assets 

### LINKED ISSUE #46: Add good caching headers for assets (non-dynamic files)
Tobira uses (and will use) completely static files that should be cached by the browser to keep page loads as short as possible. In the backend, we know best how long some things can/should be cached and we should therefore add corresponding HTTP headers there IMO. Passing the responsibility to the server admin and forcing an nginx reverse proxy seems like a bad idea to me. 

Some thoughts:

- The initial HTML request (`index.html` basically) should never be cached. Luckily that one is only loaded once per session.
- The GraphQL API should also never be cached.
- Assets should be cached. I think a common strategy is to include a short hash of an asset's contents in its filename and to set the max age to a high value. That way the assets can be cached for a very long time (think: months) and if it doesn't change, it the user never has to download it again. Whenever an asset changes, the hash and thus filename changes (which is included in the non-cached `index.html`) and consequently, the browser has to download the new version.

### LINKED ISSUE #96: Do not embed dummy/test logo into production binary
The logo we have for testing purposes shouldn't be included in the final binary. Once the logo is configurable (see #75), we the test logo should be used like normal logos and not embedded.

### LINKED ISSUE #144: Improve handling of assets
We currently use `rust-embed` to embed our assets into the binary in release mode, and dynamically load in debug mode. The crate is not optimal though and we need more flexibility.

### LINKED ISSUE #145: Change CI back to building in debug mode instead of release mode
Due to how we currently include assets and since we started deploying in CI, CI now does release builds. Which takes a very long time. This should be changed back once we can better control the assets. 

Thus blocked by #144.

### PR DESCRIPTION
**Pull request by LukasKalbertodt**
Sorry, this PR has become quite large, especially considering the code in `reinda`. It sure was a ride, but I am pretty happy with the results.

Closes #144 
Closes #145 
Closes #96 
Closes #46 
Lays foundations for #20 and #75 
CC #152 

This PR replaces `rust-embed` with `reinda` to handle assets. With this, we get the flexibility we need: some assets have to be included at runtime and can't be embedded (e.g. the logo). Furthermore the filename hashing of `reinda` makes it possible to fix #46. This PR also contains a bunch of misc improvements. The commit messages contain more information.

It can be reviewed commit by commit.

(Still draft as the reinda dependency still points to my git repo instead of a released version)

**TODO**:
- [x] Release `reinda 0.2` and use that in `Cargo.toml`
- [x] Test the CI/deploy changes (deployed [here](https://betterassethandling-ymv0dgvylwfzc2v0lwhhbmrsaw5n.tobira.opencast.org/))

### CODE REVIEW

**Comment by LukasKalbertodt** [comment]
I fixed a tiny thing in my ansible script changes (amended to last commit) and tested it. The deployment works (well, there is another bug, independent of this PR).

**Review by JulianKniephoff — COMMENTED** [review]
Okay, phew, this was a big one. Surprisingly (not meant in a mean way) I don't have much to complain about, though; just a very minor thing we can discuss maybe.

But other than that: I just ran `tobira` from my Desktop by just copying four files after building it, very cool! :D

File: backend/server/src/http/assets.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        let mime_guess = mime_guess::from_path(path).first();
+        if let Some(mime) = mime_guess {
+            builder = builder.header("content-type", mime.to_string())
+        }
+
+        // Caching header if the filename contains a content hash. We can unwrap
+        // the `lookup` call as we know from above that the path is valid.
+        if self.assets.asset_info(self.assets.lookup(path).unwrap()).is_filename_hashed() {
+            // This is one year in seconds.
+            builder = builder.header("cache-control", "public, max-age=31536000");
```
Comment (JulianKniephoff) [inline backend/server/src/http/assets.rs:124]: Should we add `immutable` here? See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#caching_static_assets

Comment (LukasKalbertodt) [reply backend/server/src/http/assets.rs:124]: Seems reasonable

**Review by JulianKniephoff — APPROVED** [review]

**Pull request merged**